src/Pages/ArticlesByTopic.jsx

```diff
@@ -0,0 +1,45 @@
+import { useState, useEffect } from "react";
+import { getArticlesByTopic } from "../Utils/api";
+import { ArticleCard } from "../Components/ArticleCard";
+import { Link, useParams } from "react-router-dom";
+
+export const ArticlesByTopic = () => {
+  const [articles, setArticles] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const { topicSlug } = useParams();
+
+  console.log(topicSlug);
+
+  useEffect(() => {
+    setLoading(true);
+    getArticlesByTopic(topicSlug).then((articles) => {
+      const articlesByTopic = articles.filter(
+        (article) => article.topic === topicSlug
+      );
+
+      setArticles(articlesByTopic);
+      setLoading(false);
+    });
+  }, [topicSlug]);
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  return (
```

Comment:
A navigation bar would be nice here, so the user can see articles from other topics as well starting from this page.